frappe/integrations/doctype/google_drive/google_drive.py

```diff
@@ -0,0 +1,256 @@
+				"redirect_uri": redirect_uri,
+				"grant_type": "authorization_code"
+			}
+			r = requests.post(get_auth_url(), data=data).json()
+
+			if "refresh_token" in r:
+				frappe.db.set_value("Google Drive", google_drive.name, "refresh_token", r.get("refresh_token"))
+				frappe.db.commit()
+
+			frappe.local.response["type"] = "redirect"
+			frappe.local.response["location"] = "/desk#Form/{0}".format(quote("Google Drive"))
+
+			frappe.msgprint(_("Google Drive has been configured."))
+		except Exception as e:
+			frappe.throw(e)
+
+def get_authentication_url(client_id, redirect_uri):
+	return {
+		"url": "https://accounts.google.com/o/oauth2/v2/auth?access_type=offline&response_type=code&prompt=consent&client_id={}&include_granted_scopes=true&scope={}&redirect_uri={}".format(client_id, SCOPES, redirect_uri)
+	}
+
+@frappe.whitelist()
+def google_callback(code=None):
+	"""
+		Authorization code is sent to callback as per the API configuration
+	"""
+	frappe.db.set_value("Google Drive", None, "authorization_code", code)
+	frappe.db.commit()
+
+	authorize_access()
+
+def get_google_drive_object():
+	"""
+		Returns an object of Google Drive.
+	"""
+	google_settings = frappe.get_doc("Google Settings")
+	account = frappe.get_doc("Google Drive")
+
+	credentials_dict = {
+		"token": account.get_access_token(),
+		"refresh_token": account.get_password(fieldname="refresh_token", raise_exception=False),
+		"token_uri": get_auth_url(),
+		"client_id": google_settings.client_id,
+		"client_secret": google_settings.get_password(fieldname="client_secret", raise_exception=False),
+		"scopes": "https://www.googleapis.com/auth/drive/v3"
+	}
+
+	credentials = google.oauth2.credentials.Credentials(**credentials_dict)
+	google_drive = googleapiclient.discovery.build("drive", "v3", credentials=credentials)
+
+	return google_drive, account
+
+def check_for_folder_in_google_drive():
+	"""Checks if folder exists in Google Drive else create it."""
+	def _create_folder_in_google_drive(google_drive, account):
+		file_metadata = {
+			"name": account.backup_folder_name,
+			"mimeType": "application/vnd.google-apps.folder"
+		}
+
+		try:
+			folder = google_drive.files().create(body=file_metadata, fields="id").execute()
+			frappe.db.set_value("Google Drive", None, "backup_folder_id", folder.get("id"))
+			frappe.db.commit()
+		except HttpError as e:
+			frappe.throw(_("Google Drive - Could not create folder in Google Drive - Error Code {0}").format(e))
+
+	google_drive, account = get_google_drive_object()
+
+	if account.backup_folder_id:
+		return
+
+	backup_folder_exists = False
+
+	try:
+		google_drive_folders = google_drive.files().list(q="mimeType='application/vnd.google-apps.folder'".format(account.backup_folder_name)).execute()
+	except HttpError as e:
+		frappe.throw(_("Google Drive - Could not find folder in Google Drive - Error Code {0}").format(e))
+
+	if google_drive_folders.get("files"):
```

Comment:
i suppose this check isn't needed, the next line will default to `None` for `get()`